tools/SetupFlow/DevHome.SetupFlow/Services/WindowsPackageManager.cs

```diff
@@ -41,15 +43,18 @@ public class WindowsPackageManager : IWindowsPackageManager
     public WindowsPackageManager(
         WindowsPackageManagerFactory wingetFactory,
         IAppInstallManagerService appInstallManagerService,
-        IPackageDeploymentService packageDeploymentService)
+        IPackageDeploymentService packageDeploymentService,
+        IHost host)
     {
         _wingetFactory = wingetFactory;
         _appInstallManagerService = appInstallManagerService;
         _packageDeploymentService = packageDeploymentService;
 
+        var activityId = host.GetService<SetupFlowOrchestrator>().ActivityId;
+
         // Lazy-initialize custom composite catalogs
-        _allCatalogs = new (CreateAllCatalogs);
-        _wingetCatalog = new (CreateWinGetCatalog);
+        _allCatalogs = new (CreateAllCatalogs(activityId));
```

Comment:
Those values are Lazy inside the singleton, hence they will not change after the object is first initialized

Reply:
Removed.